lemminx-maven/src/main/java/org/eclipse/lemminx/maven/PluginValidator.java

```diff
@@ -84,12 +85,105 @@ private static Diagnostic internalValidateConfiguration(DiagnosticRequest diagno
+				validateConfigurationType(diagnosticReq, parameter);
 				return null;
 			}
 		}
-		return new Diagnostic(diagnosticReq.getRange(), "Invalid plugin configuration: " + diagnosticReq.getCurrentTag(), DiagnosticSeverity.Warning,
+		return new Diagnostic(diagnosticReq.getRange(),
+				"Invalid plugin configuration: " + diagnosticReq.getCurrentTag(), DiagnosticSeverity.Warning,
 				diagnosticReq.getXMLDocument().getDocumentURI(), "XML");
 
 	}
 
+	private static void validateConfigurationType(DiagnosticRequest diagnosticReq, Parameter parameter) {
+		DOMNode node = diagnosticReq.getNode();
+		System.out.println("Processing: " + node.getLocalName());
+		System.out.println("Children: ");
+		node.getChildren().forEach(n -> System.out.println("local name: " + n.getLocalName()));
+		node.getChildren().forEach(n -> System.out.println("value: " + n.getNodeValue()));
+		// TODO: Type needs to be parsed to account for array '[] 'string
+		String type = parameter.getType();
+		if (!node.hasChildNodes()) {
+			if (parameter.isRequired()) {
+				diagnosticReq.getDiagnostics()
+						.add(new Diagnostic(diagnosticReq.getRange(),
+								"Missing required parameter: " + parameter.getName(), DiagnosticSeverity.Warning,
+								diagnosticReq.getXMLDocument().getDocumentURI(), "XML"));
+
+			}
+			return;
+		}
+
+		if (type.endsWith("[]")) {
+			// should have child nodes which are tags
+			if (!node.getChildren().stream().allMatch(n -> n.isElement())) {
+				diagnosticReq.getDiagnostics()
+						.add(new Diagnostic(diagnosticReq.getRange(),
+								"This configuration parameter requires a child tag", DiagnosticSeverity.Warning,
+								diagnosticReq.getXMLDocument().getDocumentURI(), "XML"));
+				return;
+			}
+
+			for (DOMNode childNode : node.getChildren()) {
+				DiagnosticRequest childDiagnosticReq = new DiagnosticRequest(childNode, diagnosticReq.getXMLDocument(),
+						diagnosticReq.getDiagnostics());
+				System.out.println("Processing child node: " + childNode.getLocalName());
+					internalValidateConfigurationType(childDiagnosticReq, type.substring(0, type.length() - 2));					
+
+			}
+		} else {
+			internalValidateConfigurationType(diagnosticReq, type);
+		}
+
+	}
+
+	private static void internalValidateConfigurationType(DiagnosticRequest diagnosticReq, String type) {
+		DOMNode node = diagnosticReq.getNode();
+		if (!node.hasChildNodes()) {
+			return;
+		}
+		if (node.getChildren().size() > 1) {
+			for (DOMNode childNode : node.getChildren()) {
+				DiagnosticRequest childDiagnosticReq = new DiagnosticRequest(childNode, diagnosticReq.getXMLDocument(),
+						diagnosticReq.getDiagnostics());
+				System.out.println("Processing child node: " + childNode.getLocalName());
+				internalValidateConfigurationType(childDiagnosticReq, type);			
+			}
+		}
+		
+		DOMNode childNode = node.getChild(0);
+		if (childNode.isComment()) {
+			return;
+		}
+		
+		String value = childNode.getNodeValue();
+		switch (type) {
+		case "java.lang.Integer":
```

Comment:
The limitation is that the type might be a custom interface defined in the MOJO/plugin, such is the case for [maven-enforcer-plugin](https://github.com/apache/maven-enforcer/blob/master/enforcer-api/src/main/java/org/apache/maven/enforcer/rule/api/EnforcerRule.java)